### PR TITLE
How to debug an iOS app

### PR DESCRIPTION
**Pull request by marcmolla**
I created a tutorial for the remote debugging. Not sure if Toga project is the right place, as debugging is generic, but others options did not convince me...

### CODE REVIEW

**Comment by Ocupe** [comment]
I was looking for a solution like that yesterday. Thanks for sharing!

File: docs/howto/remote-debugging.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
+======================================
+How to debug an app in iOS and Android
```
Comment (eliasdorneles) [inline docs/howto/remote-debugging.rst]: I feel this title should specify that this tutorial is Pycharm and iOS only: `How to debug an iOS app from PyCharm`

Note: I'm not a toga maintainer nor anything, just a random internet person looking through the issues :)

Comment (freakboy3742) [inline docs/howto/remote-debugging.rst]: Agreed that this should be flagged as a "Pycharm" tutorial - especially given that one of the longer-term goals (and the original reason BeeWare started) is to develop a debugger :-)

**Comment by phildini** [comment]
Hi there! We're closing this at this point due to inactivity, but feel free to re-open if you're interested in continuing it! Thanks!

**Comment by BrendanSimon** [comment]
Anyone know if a similar thing can be achieved with VS Code and `ptvsd` (Python Tools for Visual Studio Debug server) ?

https://github.com/microsoft/ptvsd

**Comment by freakboy3742** [comment]
No idea - but it would be interesting to find out!

**Comment by BrendanSimon** [comment]
Well I got half way there.  I can debug a Mac OS X toga app ok with the `ptvsd` extension.

I followed the following instructions for `pydev` and `PyCharm` as a guide, modifying things for `ptvsd`.

https://github.com/marcmolla/toga/blob/upstream/docs/howto/remote-debugging.rst

Modification of `__main__.py` for `examples/detailedlist/`

```python
if __name__ == '__main__':
    import ptvsd
    ptvsd.enable_attach( address = ('0.0.0.0', 5678) )
    ptvsd.wait_for_attach()
    # breakpoint()

    main().main_loop()
```

The app seems to stop in the simulator waiting for the debugger to attach, but when it attempts to attach I get error messages from `ptvsd`

```
connect ECONNREFUSED 127.0.0.1:5678
```

I've just copied the `ptvsd` directory from my venv to the app bundle.

```
cp -a ../../venv3.7//lib/python3.7/site-packages/ptvsd iOS/DetailedList\ Demo/detailedlist/app_packages/
```

I hope that is enough and that it is a pure python module.  It doesn't get packaged by default by briefcase, but maybe that's because the `import ptvsd` is guarded by the `if __name__ == '__main__:'` statement?  I can try with it at the file scope level.

Anyway, there may be something else going on, as I've just tried commenting out all the `ptvsd` stuff and the app isn't running in the simulator, so I must have broken something else in the process -- e.g. I've installed toga with `pip install -e` as I'm trying to implement `toga.OptionContainer` on iOS, thus the need for some debugging tools :)

**Comment by BrendanSimon** [comment]
Looks like something is screwy with my toga git repo or venv as I can't get any demos or examples to run on iOS simulator.

I have my own toga app project which just uses toga boxes, labels and scroll list, which does run in iOS simulator ok (well after I had to delete the iOS directory as I was getting a build error - I'm assuming after an OS X Catalina, XCode, Simulator upgrade).

The app waited to attach, but didn't hit the breakpoint I set on first line of app.  The app just ran as per normal (after debugger attach).

I enabled the `breakpoint()` statement and ran again.  It hit the breakpoint but couldn't resolve the source file.  I got the following error message in vscode popup.

```
Unable to open '__main__.py': Unable to read file '/Users/brendan/Sandbox/sggc_live/sggclive/Users/brendan/Library/Developer/CoreSimulator/Devices/6C3D8FBB-BC84-4972-A77C-4B2CE2BF1720/data/Containers/Bundle/Application/74B67E5A-5D18-4C28-B9B8-F5B9422F8799/sggclive.app/Library/Application Support/au.com.etrix.sggclive.sggclive/app/sggclive/__main__.py' (Error: Unable to resolve non-existing file '/Users/brendan/Sandbox/sggc_live/sggclive/Users/brendan/Library/Developer/CoreSimulator/Devices/6C3D8FBB-BC84-4972-A77C-4B2CE2BF1720/data/Containers/Bundle/Application/74B67E5A-5D18-4C28-B9B8-F5B9422F8799/sggclive.app/Library/Application Support/au.com.etrix.sggclive.sggclive/app/sggclive/__main__.py').
```

Seems to be a path issue in the build.  Is this a briefcase issue/bug ?

Getting closer !!

**Pull request closed without merging**